### DISCUSSION TITLE
Save file

### DISCUSSION THREAD

**Discussion by duksosleepy** [pymupdf/PyMuPDF#1797] [Q&A] [answered]
When i save the page with some change, module create a new file. Can I save this change in original file ?

**Comment by JorjMcKie** [comment, marked as answer]
This is a typical "Discussions" post - by no means an issue.

You have two options:
1. Saving to a new file. Here you have all variants for changing compression and setting other parameters.
2. Saving to the original. This requires setting "incremental" to `True`. Other parameters are largely excluded when using this option.

Please tell us what was unclear in the documentation about `Document.save()` - so we can improve the description.

**Reply by duksosleepy** [reply]
Hmm. How to use save() with incremental parameter?

**Reply by JorjMcKie** [reply]
`doc.save(incremental=True` or `doc.saveIncr()`

**Reply by duksosleepy** [reply]
It worked with doc.saveIncr() command, but doc.save(incremental=True) command raise exception "incremental needs original file"

**Reply by JorjMcKie** [reply]
please study the documentation

**Reply by duksosleepy** [reply]
may be need a doc.name parameter ?

**Reply by duksosleepy** [reply]
Thank you so much !!!
